libraries/AP_Compass/AP_Compass.h

```diff
@@ -273,11 +273,20 @@ friend class AP_Compass_Backend;
     const Vector3f &get_motor_offsets(uint8_t i) const { return _get_state(Priority(i)).motor_offset; }
     const Vector3f &get_motor_offsets(void) const { return get_motor_offsets(_first_usable); }
 
-    /// Set the throttle as a percentage from 0.0 to 1.0
+    /// Set the throttle at the time the battery was read as a percentage from 0.0 to 1.0
     /// @param thr_pct              throttle expressed as a percentage from 0 to 1.0
     void set_throttle(float thr_pct) {
-        if (_motor_comp_type == AP_COMPASS_MOT_COMP_THROTTLE) {
-            _thr = thr_pct;
+        _read_thr = thr_pct;
+        _thr_mot_factor = 1.0f;
+    }
+
+    /// Set the current throttle as a percentage from 0.0 to 1.0
+    /// @param thr_pct              throttle expressed as a percentage from 0 to 1.0
+    void set_current_throttle(float thr_pct) {
+        if (is_zero(_read_thr)) {
+            _thr_mot_factor = 1.0f;
+        } else {
+            _thr_mot_factor = thr_pct / _read_thr;
```

Comment:
This should probably be constrained to avoid infinities from floats, and because there is some filtering happening in the battery monitor level (IE extreme throttle punches can produce a pretty large result, and will give you a funky looking result right as you disarm I'd expect.). (Scaling up from system idle load with motors not spinning and then spinning is also not a direct multimplier on the idle load)

Reply:
Thanks, fixed